src/main/starlark/builtins_bzl/common/objc/objc_common.bzl

```diff
@@ -132,7 +132,9 @@ def _create_context_and_provider(
         )
         sdk_linking_info["sdk_dylib"].extend(compilation_attributes.sdk_dylibs.to_list())
 
-        objc_compilation_context_kwargs["public_hdrs"].extend(compilation_attributes.hdrs.to_list())
+        objc_compilation_context_kwargs["public_hdrs"].extend(
+            [artifact for artifact, _ in compilation_attributes.hdrs.to_list()],
+        )
```

Comment:
So what you are doing here is the corresponding operation to what's happening in [CompilationAttributes.Builder.addHeadersFromRuleContext()](https://github.com/bazelbuild/bazel/blob/master/src/main/java/com/google/devtools/build/lib/rules/objc/CompilationAttributes.java#L194), right? The picking of the key from the artifact/label pairs returned by [CcCommon.getHeaders()](https://github.com/bazelbuild/bazel/blob/master/src/main/java/com/google/devtools/build/lib/rules/cpp/CcCommon.java#L150)?

Unless I'm mistaken, that effectively moves this further down in the chain, changing the content of compilation_attributes.hdrs from artifacts to artifact/label pairs. And since compilation_attributes in compilation_support.bzl is not just passed to objc_common.create_context_and_provider(), but also returned, both by build_common_variables() and register_compile_and_archive_actions_for_j2objc(), so the change doesn't look contained within compilation_support.bzl.

Wouldn't it be better to make the conversion from artifact/label pairs to artifact right when hdrs is initialized in _create_compilation_attributes()? Or am I missing something?

Reply:
hrm yea good point, pushed that. I assume long term once we can drop this objc specific stuff we'll want the pairs version, but I think you're right it's nicer to not mess with now